st2common/st2common/services/packs.py

```diff
@@ -290,3 +293,112 @@ def delete_action_files_from_pack(pack_name, entry_point, metadata_file):
+    dest_pack_base_path,
+    dest_pack,
+    dest_action,
+):
+    """
+    Prepares the path for entry point and metadata files for source and destination.
+    Clones the content from source action files to destination action files.
+    """
+
+    source_metadata_file_path = os.path.join(
+        source_pack_base_path, source_metadata_file
+    )
+    dest_metadata_file_name = "%s.yaml" % (dest_action)
+    dest_metadata_file_path = os.path.join(
+        dest_pack_base_path, "actions", dest_metadata_file_name
+    )
+    _clone_content_to_destination_file(
+        source_file=source_metadata_file_path, destination_file=dest_metadata_file_path
+    )
+
+    if source_entry_point:
+        if source_runner_type in ["orquesta", "action-chain"]:
+            dest_entry_point_file_name = "workflows/%s.yaml" % (dest_action)
+            # creating workflows directory if doesn't exist
+            wf_dir_path = os.path.join(dest_pack_base_path, "actions", "workflows")
+            if not os.path.isdir(wf_dir_path):
+                os.mkdir(path=wf_dir_path)
+        else:
+            old_ext = os.path.splitext(source_entry_point)[1]
+            dest_entry_point_file_name = dest_action + old_ext
+
+        source_entry_point_file_path = os.path.join(
+            source_pack_base_path, "actions", source_entry_point
+        )
+        dest_entrypoint_file_path = os.path.join(
+            dest_pack_base_path, "actions", dest_entry_point_file_name
+        )
+        _clone_content_to_destination_file(
+            source_file=source_entry_point_file_path,
+            destination_file=dest_entrypoint_file_path,
+        )
+    else:
+        dest_entry_point_file_name = ""
+
+    with open(dest_metadata_file_path) as df:
+        doc = yaml.load(df, Loader=yaml.FullLoader)
+
+    doc["name"] = dest_action
+    if "pack" in doc:
+        doc["pack"] = dest_pack
+    doc["entry_point"] = dest_entry_point_file_name
+
+    with open(dest_metadata_file_path, "w") as df:
+        yaml.dump(doc, df, default_flow_style=False, sort_keys=False)
+
+
+def clone_action_db(source_action_db, dest_pack, dest_action):
+    dest_action_db = copy.deepcopy(source_action_db)
+    source_runner_type = source_action_db["runner_type"]["name"]
+    if source_action_db["entry_point"]:
+        if source_runner_type in ["orquesta", "action-chain"]:
+            dest_entry_point_file_name = "workflows/%s.yaml" % (dest_action)
+        else:
+            old_ext = os.path.splitext(source_action_db["entry_point"])[1]
+            dest_entry_point_file_name = dest_action + old_ext
+    else:
+        dest_entry_point_file_name = ""
+    dest_action_db["entry_point"] = dest_entry_point_file_name
+    dest_action_db["metadata_file"] = "actions/%s.yaml" % (dest_action)
+    dest_action_db["name"] = dest_action
+    dest_ref = ".".join([dest_pack, dest_action])
+    dest_action_db["ref"] = dest_ref
+    dest_action_db["uid"] = "action:%s:%s" % (dest_pack, dest_action)
```

Comment:
Instead of using colon, use the UID separator `UIDFieldMixin.UID_SEPARATOR` and import `from st2common.models.db.stormbase import UIDFieldMixin`. If there's any change to the separator symbol, this won't be broken.

```
```